### PR TITLE
add test for targetPathComputationStrategy and partialUpdate

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy [hunk 1]
```diff
@@ -48,6 +49,18 @@ class PartialUpdateSpec extends HealthCheckSpecification {
                 [
                         field   : "priority",
                         newValue: 654
+                ],
+                [
+                        field   : "maxLatency",
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy:54]: already tested above

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy [hunk 2]
```diff
@@ -102,4 +107,74 @@ class PathComputationSpec extends HealthCheckSpecification {
...
+
+        when: "Sync the flow"
+        northbound.synchronizeFlow(flow.flowId)
+
+        then: "Path computation strategy is updated and targetPathComputationStrategy is deleted"
+        with(northbound.getFlow(flow.flowId)) {
+            pathComputationStrategy == latencyStrategy
+            !targetPathComputationStrategy
+        }
+
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy:145]: end of test1
start of test2

We should not overuse long scenarios. Moreover in this case scenarios a pretty much isolated and don't depend on each other, don't see any benefits in keeping them in 1 test

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy [hunk 3]
```diff
@@ -102,4 +107,74 @@ class PathComputationSpec extends HealthCheckSpecification {
...
+        when: "Update path computation strategy(latency -> cost) via partialUpdate"
+        northbound.partialUpdate(flow.flowId, new FlowPatchDto().tap {
+            it.targetPathComputationStrategy = costStrategy
+        })
+        Wrappers.wait(WAIT_OFFSET / 2) { assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP }
+
+        and: "Reroute the flow"
+        northboundV2.rerouteFlow(flow.flowId)
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy:153]: What if latency and cost paths are the same? Will the update actually happen?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy:145]: done

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy [hunk 4]
```diff
@@ -102,4 +107,88 @@ class PathComputationSpec extends HealthCheckSpecification {
...
+        def latencyStrategy = PathComputationStrategy.LATENCY.toString().toLowerCase()
+        def costStrategy = PathComputationStrategy.COST.toString().toLowerCase()
+        def flow = flowHelperV2.randomFlow(swPair).tap { it.pathComputationStrategy = costStrategy }
+        flowHelperV2.addFlow(flow)
+
+        when: "Update path computation strategy(cost -> latency) via partialUpdate"
+        northbound.partialUpdate(flow.flowId, new FlowPatchDto().tap {
+            it.targetPathComputationStrategy = latencyStrategy
+        })
+        Wrappers.wait(WAIT_OFFSET / 2) { assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP }
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy:126]: Partial update of targetPathComputationStrategy should not cause a flow status change. Can you recheck this place please? I think these waits can be removed

Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy]: fixed

**Pull request merged**